### DISCUSSION TITLE
Temporarily disable starship?

### DISCUSSION THREAD

**Discussion by shoffmeister** [starship/starship#4408] [Q&A]
What is the best approach to _temporarily_ disable `starship` and get a distro default prompt, in my case on Fedora Linux 36 with bash?

Context is recording a screen cast (via `asciinema` or `peek`) where I do not want the `starship` output to show up - but the bland out-of-the-box distro bash prompt.

It would perhaps feel natural to have a "Prompt" configuration option such as "Disabled = true" or similar, which could be conveniently used to toggle all of `starship` on the fly? Alas, I don't see such a configuration option?

**Comment by chipbuster** [comment]
starship hooks several variables which are tricky to unset. This means that once starship is set, it's very difficult to revert to the default prompt without shell-specific code (meaning that a global disable flag is tricky to support in starship).

I use something like this in my bashrc for testing purposes, perhaps it might be useful to you:

```
if [[ "$NOSTARSHIP" != "1" ]]; then
    eval "$(starship init bash)"
fi
```

You can then run the following command to start up a bash shell without executing the starship hooks:

```
NOSTARSHIP=1 bash
```

**Reply by shoffmeister** [reply]
Many thanks for the insight - this makes a whole lot of sense. I have opted for
```
#
# Install starship into the new shell, but gated on the
# presence of a (temporary) disabling environment variable.
#
# Use
#
#   SUPPRESS_STARSHIP=true bash
#
# to launch a shell without starship, for instance for
# screen casting.
#
if [[ -z "${SUPPRESS_STARSHIP}" ]]; then
        eval "$(starship init bash)"
fi
```
